ui/src/components/ClientForm/index.jsx

```diff
@@ -184,6 +190,11 @@ export default class ClientForm extends Component {
       expandedScopes,
       disabled,
     } = this.state;
+    const isClientDirty =
+      clientId !== client.ID ||
+      description !== client.description ||
+      expires !== client.expires ||
+      scopeText !== client.scopeText;
```

Comment:
`clientId` is not editable when viewing an existent client so it should be okay to omit that. We also need to make sure this code still works when a user is creating a new client (i.e., when `client === null`). I think we can do the following change:

```jsx
const isClientDirty =
  isNewClient ||
  description !== client.description ||
  expires !== client.expires ||
  scopeText !== client.scopes.join('\n');
```

Note: It's `client.scopes`, not `client.scopeText`